python/coinbase-agentkit/tests/action_providers/weth/test_wrap_eth.py

```diff
@@ -0,0 +1,179 @@
+"""Tests for WETH action provider."""
+from unittest.mock import patch
+
+import pytest
+from pydantic import ValidationError
+
+from coinbase_agentkit.action_providers.weth.constants import MIN_WRAP_AMOUNT
+from coinbase_agentkit.action_providers.weth.schemas import WrapEthInput
+from coinbase_agentkit.action_providers.weth.weth_action_provider import (
+    WETH_ABI,
+    WETH_ADDRESS,
+    WethActionProvider,
+)
+from coinbase_agentkit.network import Network
+
+# Import constants from conftest
+from .conftest import (
+    MOCK_ADDRESS,
+    MOCK_NETWORK,
+    MOCK_RECEIPT,
+    MOCK_TX_HASH,
+)
+
+# Test constants
+MOCK_AMOUNT = str(MIN_WRAP_AMOUNT)
+
+def test_wrap_eth_input_model_valid():
+    """Test that WrapEthSchema accepts valid parameters."""
+    input_model = WrapEthInput(amount_to_wrap=MOCK_AMOUNT)
+
+    assert isinstance(input_model, WrapEthInput)
+    assert input_model.amount_to_wrap == MOCK_AMOUNT
+
+def test_wrap_eth_input_model_invalid_format():
+    """Test that WrapEthSchema rejects invalid format inputs."""
+    invalid_inputs = [
+        "",
+        "-123",
+        "abc",
+        "123.456",
+        "123abc",
+    ]
+    for invalid_input in invalid_inputs:
+        with pytest.raises(ValidationError) as exc_info:
+            WrapEthInput(amount_to_wrap=invalid_input)
+        assert "Amount must be a whole number as a string" in str(exc_info.value)
+
+def test_wrap_eth_input_model_below_minimum():
+    """Test that WrapEthSchema rejects amounts below minimum."""
+    below_min = str(MIN_WRAP_AMOUNT - 1)
+    with pytest.raises(ValidationError) as exc_info:
+        WrapEthInput(amount_to_wrap=below_min)
+    assert f"Amount must be at least {MIN_WRAP_AMOUNT} wei" in str(exc_info.value)
+
+def test_wrap_eth_input_model_missing_params():
+    """Test that WrapEthSchema raises error when params are missing."""
+    with pytest.raises(ValidationError):
+        WrapEthInput()
+
+def test_wrap_eth_success():
+    """Test successful ETH wrapping."""
+    with (
+        patch("coinbase_agentkit.action_providers.weth.weth_action_provider.Web3") as mock_web3,
+        patch("coinbase_agentkit.wallet_providers.EvmWalletProvider") as mock_wallet,
+    ):
+        # Set up mocks
```

Comment:
can we remove the extra comments in the tests. They are not necessary. Let's only include comments outside of doc strings that are necessary for explaining difficult code